state/account.go

```diff
@@ -147,6 +147,12 @@ func (st *Account) AccountType() int32 {
 
 // SetPendingNonce sets the pending nonce
 func (st *Account) SetPendingNonce(nonce uint64) error {
+	// this is a legacy account that had never initiated an outgoing transaction
+	// so we can convert it to zero-nonce account
+	if st.accountType == 0 && st.nonce == 0 && nonce == 1 {
+		st.accountType = 1
+	}
```

Comment:
if SetPendingNonce(2) it will skip this conversion, and remain legacy account.
Do we want to allow this behavior? i.e., the legacy account could convert to zero-nonce type (by sending first tx nonce=0), or remain legacy (by sending first tx nonce=1)

Reply:
the account isn't a fresh account if its pending nonce is set to 2